core/src/bundle_stage.rs

```diff
@@ -1307,13 +1349,46 @@ impl BundleStage {
             r_poh_recorder.would_be_leader(DROP_BUNDLE_SLOT_OFFSET * DEFAULT_TICKS_PER_SLOT);
         drop(r_poh_recorder);
 
+        let last_slot = bundle_stage_leader_stats.current_slot;
         bundle_stage_leader_stats.maybe_report(id, &working_bank_start);
 
         match (working_bank_start, would_be_leader_soon) {
             // leader now, insert new read bundles + as many as can read then return bank
             (Some(bank_start), _) => {
                 consensus_cache_updater.maybe_update(&bank_start.working_bank);
 
+                // If slot changed and currently leader, reserve space
+                let is_new_slot = match (last_slot, bundle_stage_leader_stats.current_slot) {
```

Comment:
I think we'll want a better way to be notified of new slots. Right now this only gets triggered when we receive bundles, but banking_stage could be well into using up all the CUs by the time this receives a bundle and reserves space

Reply:
this get's called every SLOT_BOUNDARY_CHECK_PERIOD whether there are bundles to process or not
https://github.com/jito-foundation/jito-solana/blob/ce881f5413cb319d8e55fc3b44d01bcb0289e38c/core/src/bundle_stage.rs#L1402-L1408

it's kinda weird flow, but i was trying to make minimal changes to what's there

Reply:
currently, every 10 ms